rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs

```diff
@@ -205,3 +201,225 @@ impl InteropIdRange {
+            interop_id_range.count,
+        );
+        assert_eq!(compressor.normalize_to_op_space(-1 as f64), 0 as f64);
+        assert_eq!(compressor.normalize_to_op_space(-5 as f64), 4 as f64);
+        let new_final = compressor.generate_next_id();
+        assert_eq!(compressor.normalize_to_op_space(new_final), new_final);
+
+        assert!(compressor.normalize_to_op_space(-6 as f64).is_nan());
+        assert_eq!(
+            compressor.error_string,
+            Some(String::from(
+                NormalizationError::UnknownSessionSpaceId.get_error_string()
+            ))
+        );
+    }
+
+    #[test]
+    fn normalize_to_session_space() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        _ = compressor.finalize_range(
+            interop_id_range.token,
+            interop_id_range.local,
+            interop_id_range.count,
+        );
+        assert_eq!(
+            compressor.normalize_to_session_space(0 as f64, 1.0),
+            -2 as f64
+        );
+        assert!(compressor
+            .normalize_to_session_space(3 as f64, -1.0)
+            .is_nan());
+        assert_eq!(
+            compressor.error_string,
+            Some(String::from(
+                SessionTokenError::UnknownSessionToken.get_error_string()
+            ))
+        );
+        assert!(compressor
+            .normalize_to_session_space(0 as f64, 7.0)
+            .is_nan());
+    }
+
+    #[test]
+    #[should_panic]
+    fn decompress_invalid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.decompress(1.0);
+    }
+
+    #[test]
+    fn decompress() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
+            compressor.generate_next_id();
+        }
+        let interop_id_range = compressor.take_next_range();
+        _ = compressor.finalize_range(
+            interop_id_range.token,
+            interop_id_range.local,
+            interop_id_range.count,
+        );
+        assert!(compressor.decompress(0.0).is_ok());
+        assert!(compressor.decompress(4.0).is_ok());
+    }
+    #[test]
+    #[should_panic]
+    fn recompress_invalid_uuid_string() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.recompress(String::from("invalid_uuid"));
+    }
+
+    #[test]
+    #[should_panic]
+    fn recompress_unknown_uuid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.recompress(String::from("0002c79e-b536-4776-b000-000266c252d5"));
+    }
+
+    #[test]
+    fn recompress() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        for _ in 0..5 {
```

Comment:
this code (generating 5 ids, finalizing range) is repeated a bunch. let's extract it; you could even have it return a vec of all the IDs and use it in the tests that want to capture them